examples/directchat.nim

```diff
@@ -49,8 +49,8 @@ type
 
 proc id (p: ChatProto): string =
   result = "unknown"
-  if p.conn.peerInfo.peerId.isSome:
-    result = $p.conn.peerInfo.peerId.get()
+  if p.conn.peerInfo.isSome:
```

Comment:
something to think of in compiled languages like nim: it's better not to do string assignments unnecessarily, so:
```
proc id(..): string = 
  if p.conn.peerInfo.isSome: ... 
  else: "unknown"
```

Reply:
Is there really a big difference with regards to interpreted languages when it comes to allocations?

In general, I've seen advocates of this style on both sides of the isle - interpreted and compiled. The argument for it is that it makes the code more readable and less error prone; against, as you imply, the _potentialy_ redundant allocation. 

I think you can argue the pros and cons of this style until the end of times and end up right where you started - as it usually happens with this sort of things. Personally, I've gotten quite used to it and I do find it more readable and less error prone, in addition moder optimizers are good enough to spot optimize it out altogether - this is also partially confirmed by inspecting the generated C code, the string is nowhere to be found.

That said, I'm neither agreeing nor disagreeing - IMO this is a matter of style/taste.

Reply:
The Nim compiler in particular doesn't have any optimisation pass that will help here. The C compiler may be able to figure it out, but considering that copying a string invokes the side-effects of the entire memory allocator (this includes potentially calling an OS routine such as `mmap`), the compiler should be shy about making this transformation.

Reply:
Ok, seems like the safest thing to do here is to avoid this style.